storage/mutable_segment.go

```diff
@@ -186,18 +186,21 @@ func (s *mutableSeg) QueryRaw(
 		return nil, errMutableSegmentAlreadySealed
 	}
 
+	numDocuments := s.mutableSegmentBase.NumDocuments()
+	if numDocuments == 0 {
+		s.RUnlock()
+		return q.NewRawResults(), nil
```

Comment:
without this, code continues to execute until `applyFilters` which will fail and throw: `no time values in timestamp field` (no docs => no timestamp index)

Reply:
Yeah fair enough.